File: Neetcode250/arrays_hashing/Two_sum.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
	public:
		vector<int> twoSum(vector<int>& nums, int target) {
			unordered_map<int, int> umap;
			int complement;
			for (int i = 0; i < nums.size(); i++){
				complement = target - nums[i];
				if (umap.find(complement) != umap.end()){
					return {umap[complement], i};
				}
				umap[nums[i]] = i;
			}
			return {};
		}
};

File: Neetcode250/arrays_hashing/Two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dictionary = {}
        for index, value in enumerate(nums):
            dictionary[index] = value
        
        for key, value in dictionary.items():
            complement = target - value
            j = list(dictionary.keys())[list(dictionary.values()).index(complement)]
            if j:
                return key, j